packages/ui/src/components/ConfirmationCode/ConfirmationCode.tsx

```diff
@@ -0,0 +1,111 @@
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { PatternInput } from "@/components/PatternInput";
+import { isNaN, range } from "lodash";
+import { useToast } from "@/components/Toast";
+import { useFormatMessage } from "@/hooks";
+
+export const ConfirmationCode: React.FC<{
+  /**
+   * validation occurs once the last number written when true
+   * validation occurs on clicking on button when false
+   */
+  autoValidate?: boolean;
+  confirmCode: (newCode: number, auto: boolean) => void;
+}> = ({ autoValidate = false, confirmCode }) => {
+  const toast = useToast();
+  const intl = useFormatMessage();
+
+  const [newCode, setNewCode] = useState("");
+  const inputRefs = useRef<(HTMLInputElement | null)[]>([]);
+
+  inputRefs.current = useMemo(
+    () => range(5).map((_, index) => inputRefs.current[index] ?? null),
+    []
+  );
+
+  const handleInputChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>, index: number) => {
+      const input = e.target;
+      const code = [...newCode];
+      code[index] = input.value;
+      setNewCode(code.join(""));
+    },
+    [newCode]
+  );
+
+  // Handle backspace key
+  const handleBackspace = useCallback(
+    (e: KeyboardEvent) => {
+      if (e.key === "Backspace") inputRefs.current[newCode.length - 1]?.focus();
+    },
+    [newCode]
+  );
+
+  // Handle paste code from clipboard
+  const handlePaste = useCallback(async () => {
+    try {
+      const text = await navigator.clipboard.readText();
+      const copiedCode = text.split("");
+      const isValidCode = copiedCode.every(
+        (val) => !isNaN(val) && !isNaN(parseFloat(val))
+      );
+
+      if (!isValidCode) return;
+
+      copiedCode.forEach((num, index) => {
+        if (inputRefs.current[index])
+          inputRefs.current[index]!.value = num.toString();
+      });
+
+      setNewCode(text.slice(0, 5));
```

Comment:
It is better to move this number `5` into a constant and then reference it using this variable.